datafusion/execution/src/memory_pool/pool.rs

```diff
@@ -129,15 +189,14 @@ pub struct FairSpillPool {
 }
 
 #[derive(Debug)]
-struct FairSpillPoolState {
-    /// The number of consumers that can spill
-    num_spill: usize,
-
-    /// The total amount of memory reserved that can be spilled
-    spillable: usize,
+struct FairSpillPoolMember {
+    used: usize,
+    can_spill: bool,
+}
 
-    /// The total amount of memory reserved by consumers that cannot spill
-    unspillable: usize,
+#[derive(Debug)]
+struct FairSpillPoolState {
+    pool_members: HashMap<String, FairSpillPoolMember>,
```

Comment:
I wonder if you can keep the spill information in `FairSpillPoolState::num_spill`, etc -- that would then allow you to use the same structure to track members in both GreedyPool and FairSpillPool

Reply:
Refactored out the common logic in to `SimplePool` struct (open to suggestions on the name).